decidim-templates/spec/system/admin/admin_manages_proposal_answer_templates_spec.rb

```diff
@@ -0,0 +1,217 @@
+        expect(page).to have_current_path decidim_admin_templates.proposal_answer_templates_path
+        within ".proposal_answer-templates" do
+          expect(page).to have_i18n_content(scope_name)
+          expect(page).to have_content("My new name")
+        end
+      end
+    end
+
+    it_behaves_like "updates a template with scopes", "Global (available everywhere)"
+    it_behaves_like "updates a template with scopes", "Participatory process: A participatory process > A component"
+  end
+
+  describe "updating a template with invalid values" do
+    let!(:template) { create(:template, :proposal_answer, organization:) }
+
+    before do
+      visit decidim_admin_templates.proposal_answer_templates_path
+      click_link translated(template.name)
+    end
+
+    it "does not update the template" do
+      fill_in_i18n(
+        :proposal_answer_template_name,
+        "#proposal_answer_template-name-tabs",
+        en: "",
+        es: "",
+        ca: ""
+      )
+
+      within ".edit_proposal_answer_template" do
+        find("*[type=submit]").click
+      end
+
+      expect(page).to have_admin_callout("problem")
+    end
+  end
+
+  describe "copying a template" do
+    let!(:template) { create(:template, :proposal_answer, organization:) }
+
+    before do
+      visit decidim_admin_templates.proposal_answer_templates_path
+    end
+
+    it "copies the template" do
+      within find("tr", text: translated(template.name)) do
+        click_link "Duplicate"
+      end
+
+      expect(page).to have_admin_callout("successfully")
+      expect(page).to have_content(template.name["en"], count: 2)
+    end
+  end
+
+  describe "destroying a template" do
+    let!(:template) { create(:template, :proposal_answer, organization:) }
+
+    before do
+      visit decidim_admin_templates.proposal_answer_templates_path
+    end
+
+    it "destroys the template" do
+      within find("tr", text: translated(template.name)) do
+        accept_confirm { click_link "Delete" }
+      end
+
+      expect(page).to have_admin_callout("successfully")
+      expect(page).to have_no_i18n_content(template.name)
+    end
+  end
+
+  describe "using a proposal_answer_template" do
+    let(:participatory_process) { create(:participatory_process, title: { en: "A participatory process" }, organization:) }
+    let!(:component) { create(:component, manifest_name: :proposals, name: { en: "A component" }, participatory_space: participatory_process) }
+
+    let(:description) { "Some meaningful answer" }
+    let(:values) do
+      { internal_state: "rejected" }
+    end
+    let!(:template) { create(:template, :proposal_answer, description: { en: description }, field_values: values, organization:, templatable: component) }
+    let!(:proposal) { create(:proposal, component:) }
+
+    before do
+      visit Decidim::EngineRouter.admin_proxy(component).root_path
+      find("a", class: "action-icon--show-proposal").click
+    end
+
+    it "uses the template" do
```

Comment:
Can you add some examples about the `scope_for_availability` feature please? Like it should be available in a process but not in an assembly when it's defined for that process and things like that. Thanks

Reply:
No. I have removed the scope for availability and used a component_constraint key that uses either component_id or 0. When 0, i am reading it as a Global template ( aka available for all proposals components)

Reply:
Oh I see, that makes sense